### PR TITLE
[node] various http2 fixes

### PR DESCRIPTION
**Pull request by Flarna**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/dist/latest-v11.x/docs/api/http2.html
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by Flarna** [comment]
Please wait at least at least a day until this gets merged to allow others to review.

File: types/node/http2.d.ts [hunk 1]
```diff
@@ -268,148 +306,179 @@ declare module "http2" {
...
+        prependListener(event: string | symbol, listener: (...args: any[]) => void): this;
 
-        prependOnceListener(event: string, listener: (...args: any[]) => void): this;
         prependOnceListener(event: "altsvc", listener: (alt: string, origin: string, stream: number) => void): this;
+        prependOnceListener(event: "origin", listener: (origins: string[]) => void): this;
         prependOnceListener(event: "connect", listener: (session: ClientHttp2Session, socket: net.Socket | tls.TLSSocket) => void): this;
         prependOnceListener(event: "stream", listener: (stream: ClientHttp2Stream, headers: IncomingHttpHeaders & IncomingHttpStatusHeader, flags: number) => void): this;
+        prependOnceListener(event: string | symbol, listener: (...args: any[]) => void): this;
+
+        removeListener(event: "altsvc", listener: (alt: string, origin: string, stream: number) => void): this;
```
Comment (SimonSchick) [inline types/node/http2.d.ts]: I find it questionable to create overloads for any `removeListener` (or similar) methods as it doesn't improve type safety at all, any wrong event name of function signature will just default to the generic `removeListener(event: string, ...` variant.

For methods that take event listener for calling purposes (on, once, etc.) defining overloads makes sense at that allows argument inference.

Comment (Flarna) [reply types/node/http2.d.ts]: agree, removed all of them.

**Review by SimonSchick — APPROVED** [review]

**Comment by rbuckton** [comment]
@Flarna: Do you still need more time?

**Comment by Flarna** [comment]
@rbuckton I think it's fine to merge now.

**Pull request merged**